lib/presentation/redux/states/my_space_state.dart

```diff
@@ -39,30 +41,55 @@ import 'package:linshare_flutter_app/presentation/redux/states/linshare_state.da
 
 @immutable
 class MySpaceState extends LinShareState {
-  final List<Document> documentList;
+  final Map<Document, bool> documentList;
+  final bool isSelectingDocuments;
 
-  MySpaceState(Either<Failure, Success> viewState, this.documentList) : super(viewState);
+  MySpaceState(
+    Either<Failure, Success> viewState,
+    this.documentList,
+    this.isSelectingDocuments
+  ) : super(viewState);
 
   factory MySpaceState.initial() {
-    return MySpaceState(Right(IdleState()), []);
+    return MySpaceState(Right(IdleState()), HashMap<Document, bool>(), false);
   }
 
   @override
   LinShareState clearViewState() {
-    return MySpaceState(Right(IdleState()), []);
+    return MySpaceState(Right(IdleState()), HashMap<Document, bool>(), false);
   }
 
   @override
   LinShareState sendViewState({Either<Failure, Success> viewState}) {
-    return MySpaceState(viewState, documentList);
+    return MySpaceState(viewState, documentList, isSelectingDocuments);
   }
 
   LinShareState setDocuments({Either<Failure, Success> viewState, List<Document> newDocumentList}) {
-    return MySpaceState(viewState, newDocumentList);
+    return MySpaceState(viewState, { for (var document in newDocumentList) document: false}, isSelectingDocuments);
```

Comment:
so, all the selected document will be unselected? 
what is better for this case?

Reply:
all list will be unselected when the user is fetching new documents, I think it's a better case avoiding selected documents not existing anymore

Reply:
@dieptran88 Can you give me an UX for it? Should we disable select mode on reload request?

Reply:
i think we should not auto unselect when refreshing page, bc :